notebook/agentchat_hierarchy_flow_using_select_speaker.ipynb

```diff
@@ -0,0 +1,397 @@
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[33mA1\u001b[0m (to chat_manager):\n",
+      "\n",
+      "Yes, A2 has the value of x and A3 knows the value of y. Let me get those for you. A2, what is the value of x? And A3, what is the value of y?\n",
+      "\n",
+      "--------------------------------------------------------------------------------\n",
+      "[autogen.oai.completion: 10-31 09:38:24] {788} WARNING - Completion was provided with a config_list, but the list was empty. Adopting default OpenAI behavior, which reads from the 'model' parameter instead.\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:autogen.oai.completion:Completion was provided with a config_list, but the list was empty. Adopting default OpenAI behavior, which reads from the 'model' parameter instead.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Initialization\n",
+    "agents_A = [\n",
+    "    AssistantAgent(name='A1', \n",
+    "                   system_message=\"You are a team leader A1, your team consists of A2, A3. You can talk to the other team leader B1, whose team member is B2.\",\n",
+    "                   llm_config=llm_config),\n",
+    "    AssistantAgent(name='A2', \n",
+    "                   system_message=\"You are team member A2, you know the secret value of x but not y, x = 9. Tell others x to cooperate.\",\n",
+    "                   llm_config=llm_config),\n",
+    "    AssistantAgent(name='A3', \n",
+    "                   system_message=\"You are team member A3, You know the secret value of y but not x, y = 5. Tell others y to cooperate.\",\n",
+    "                   llm_config=llm_config)\n",
+    "]\n",
+    "\n",
+    "agents_B = [\n",
+    "    AssistantAgent(name='B1', \n",
+    "                   system_message=\"You are a team leader B1, your team consists of B2. You can talk to the other team leader A1, whose team member is A2, A3. Use NEXT: A1 to suggest talking to A1.\",\n",
+    "                   llm_config=llm_config),\n",
+    "    AssistantAgent(name='B2', \n",
+    "                   system_message=\"You are team member B2. Your task is to find out the value of x and y and compute the product. Once you have the answer, say out the answer and append a new line with TERMINATE.\",\n",
+    "                   llm_config=llm_config)\n",
+    "]\n",
+    "\n",
+    "# Terminates the conversation when TERMINATE is detected.\n",
+    "user_proxy = autogen.UserProxyAgent(\n",
+    "        name=\"User_proxy\",\n",
+    "        system_message=\"Terminator admin.\",\n",
+    "        code_execution_config=False,\n",
+    "        is_termination_msg=is_termination_msg,\n",
+    "        human_input_mode=\"NEVER\",\n",
+    "        llm_config=llm_config)\n",
```

Comment:
Does the user proxy agent need to use llm to generate reply? If not, we shouldn't set its llm_config.

Reply:
Got it, removed the llm_config from user proxy agent.